src/buffer/out/textBuffer.cpp

```diff
@@ -2935,3 +2943,32 @@ void TextBuffer::_trimMarksOutsideBuffer()
                                 }),
                  _marks.end());
 }
+
+void TextBuffer::UpdateCurrentPromptEnd(const til::point pos)

```

Comment:
Or perhaps, `UpdateCurrentMarkPromptEnd`

Or, perhaps, this is actually a _setter_ whose name should be `SetCurrentMarkPromptEnd` since it takes in a position